### DISCUSSION TITLE
Always uses defaults after reset

### DISCUSSION THREAD

**Discussion by nophead** [arendst/Tasmota#19256] [General]
Hello, I put the latest tasmota,bin on an ESP-201 module that only has 512K flash. It comes up as an AP and I can configure the Wifi SSID and password using my phone. It connects successfully to my home network and gets an IP address but then it resets. Whenever it resets it says `CFG: Use defaults` and goes back to being an access point. Why does it not remember any new settings? 

I successfully used a much older ESP-201 but that had 1MB of flash on it so I used tasmota.bin that was around at the time and had no issues with it remembering the config.

**Comment by nophead** [comment]
This is the serial log:
```
00:44:46.026 WIF: Connecting to AP MySSID as tasmota-462345-0837 ...
00:44:49.355 WIF: SSId MySSID: Connected - IP Address 192.168.1.123
00:44:59.493 APP: Restarting

 ets Jan  8 2013,rst cause:2, boot mode:(3,6)

load 0x4010f000, len 3584, room 16 
tail 0
chksum 0xb0
csum 0xb0
v3969889e
~ld

00:00:00.001 HDW: ESP8266EX
00:00:00.010 CFG: Use defaults
00:00:00.074 Project tasmota - Tasmota Version 13.0.0(lite)-2_7_4_9(2023-06-26T10:42:10)
00:00:00.227 WIF: WifiManager active for 3 minutes
00:00:00.736 HTP: Web server active on tasmota-462345-0837 with IP address 192.168.4.1
```

**Comment by barbudor** [comment]
Tasmota requires 1MB of flash. There is NO way to get it running on 512KB

Forget about 512kB modules and sincerely forget about ESP01

**Comment by nophead** [comment]
I see, thanks. I will have to try to change the memory chip then.  I made a 5 way SonOff replacement with an old ESP-201 to reduce standby power and it works fine.  I made a second one with a new ESP-201 and ran into this problem. I don't want to scrap my work because I don't think there is a pin compatible replacement. 

![image](https://github.com/arendst/Tasmota/assets/566149/9f2b5b57-c5ac-401c-b0dd-219286af57c2)

I chose the ESP-201 because it is in a metal box for safety so I need an external antenna connection.

![IMG_20230103_142724172](https://github.com/arendst/Tasmota/assets/566149/13df957a-0aa9-41b4-b5cf-5e34d6e291ef)

**Reply by sfromis** [reply]
What barbudor recommended against was ESP-01, not the different and better ESP-201 module. Should be ok to use, as long as you make sure to have at least 1MB of flash.

**Comment by nophead** [comment]
The ESP-201 I bought from the UK in 2015 For £5.47 came with 1MB flash and an antenna. Unfortunately that seller has disappeared, so I got a second  one this year from China for $5.80 and it only has 512K flash and although it said it came with an antenna it was just an antenna coax lead.

Neither advert mentioned the memory size so I assumed they were the same.

**Reply by sfromis** [reply]
When gambling on chinese sellers, you win some, and you lose some :grin: 
These days, 512KB flash chips are getting unusual, unless it is some old hardware from the dustbin....

If soldering to replace flash chips, I'd probably use 4MB flash chips. May not be much of a benefit, depending on your ideas about having fun, but same work.....

**Comment by nophead** [comment]
LOL, 4MB to control 5 sockets seems a bit excessive. In the 80's I worked at a company where we controlled 128 lamps, 16 buttons, 4 stepper motors and played sounds with 24KB of EPROM to make gaming machines.

**Reply by sfromis** [reply]
Do not count "excessive" in bytes, but in cost. Those 24KB probably cost a **lot** more.

**Reply by sfromis** [reply]
And sure, I know about compact code. I wrote a program able crack lots of C64 games in about 50-80 bytes of code.
